packages/typespec-azure-resource-manager/lib/parameters.tsp

```diff
@@ -169,3 +169,18 @@ model TenantActionScope {
   @segment("tenant")
   name: string;
 }
+
+/**
+ * Template used by ArmProviderAction templates. This produces following action route:
+ * `/{resourceUri}/providers/Microsoft.SomeRP/someAction`
+ */
+@armVirtualResource
+@extensionResource
+model ExtensionResourceActionScope {
```

Comment:
I wonder if this shouldn't just be ExtensionActionScope?  It isn't really a resource action, it's a collection action.